### PR TITLE
update swift-nio atomic use to new API

### PR DESCRIPTION
**Pull request by tomerd**
motivation: swift-nio is deprecating its AtomicBox and replaced the recommnded Atomic implementation

changes:
* use NIOAtomic instead of Atomic which is deprecated
* do not attemp to cache the connection

### CODE REVIEW

File: Sources/StatsdClient/StatsdClient.swift [hunk 1]
```diff
@@ -278,20 +279,11 @@ private final class Client {
...
         let bootstrap = DatagramBootstrap(group: self.eventLoopGroup)
             .channelOption(ChannelOptions.socket(SocketOptionLevel(SOL_SOCKET), SO_REUSEADDR), value: 1)
             .channelInitializer { channel in channel.pipeline.addHandler(Encoder(address: self.address)) }
-
         // the bind address is local and does not really matter, the remote address is addressed by AddressedEnvelope below
-        let future = bootstrap.bind(host: "0.0.0.0", port: 0)
-        future.whenSuccess { channel in
-            self.channel.store(Box(channel))
-        }
-        return future
```
Comment (tomerd) [inline Sources/StatsdClient/StatsdClient.swift:294]: removed this channel caching. could also implement a state machine instead if we think its worth while (performance)

File: Sources/StatsdClient/StatsdClient.swift [hunk 2]
```diff
@@ -240,9 +240,10 @@ private final class StatsdTimer: TimerHandler, Equatable {
 private final class Client {
     private let eventLoopGroupProvider: StatsdClient.EventLoopGroupProvider
     private let eventLoopGroup: EventLoopGroup
-    private let isShutdown = Atomic<Bool>(value: false)
+
     private let address: SocketAddress
-    private var channel: AtomicBox<Box<Channel?>> = AtomicBox(value: Box(nil))
+
+    let isShutdown = NIOAtomic<Bool>.makeAtomic(value: false)
```
Comment (Lukasa) [inline Sources/StatsdClient/StatsdClient.swift]: Did this become `internal` for a specific reason?

Comment (tomerd) [reply Sources/StatsdClient/StatsdClient.swift]: good catch thanks

**Comment by tomerd** [comment]
@Lukasa wdyt?

**Review by Lukasa — APPROVED** [review]

**Pull request merged**